xdsl/pattern_rewriter.py

```diff
@@ -435,7 +435,7 @@ class PatternRewriteWalker:
     That way, all uses are replaced before the definitions.
     """
 
-    def rewrite_module(self, op: ModuleOp):
+    def rewrite_module(self, op: (ModuleOp | Operation)):
```

Comment:
I don't agree with this change. If we actually want to use this with any Operation, the name should be changed and the type hint should be only Operation (as ModuleOp inherits from Operation). Otherwise, this should really only be called with  ModuleOp and the type hint can stay IMO.

Reply:
The issue starts here:
```
    def parse_op(self, skip_white_space: bool = True) -> Operation:
        res = self.parse_optional_op(skip_white_space=skip_white_space)
        if res is None:
            raise ParserError(self._pos, "operation expected")
        return res
```
which returns an Operation which is given as arg to rewrite_module.

Should we add something like rewrite_module_op ? or there is no point?

Reply:
I am confused. Where are these two things connected? Do we have a test that passes one into the other?

Reply:
I would also disagree in changing this. Could we also make separate PRs for this kind of things? So we can review these things separately?
If we want to make this work on any operations, I would rename this function at least.

Reply:
Since we have opened an issue for this, I am reverting this change